### PR TITLE
Rule S6969 : "memory" parameter should be specified for Scikit-Learn Pipeline

### PR DESCRIPTION
**Pull request by ghislainpiot**

### CODE REVIEW

**Review by joke1196 — CHANGES_REQUESTED** [review]
Really good job. The quickfix is super nice! I added just a few suggestions and a small question about possible FPs and test cases not covered.

**Review by joke1196 — CHANGES_REQUESTED** [review]
LGTM! You just did not address my last question/comment about the closure.

File: python-checks/src/test/resources/checks/sklearn_pipeline_specify_memory_argument.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        ...
+
+    pipeline2 = make_pipeline(StandardScaler(), LinearDiscriminantAnalysis()) # Noncompliant
+    if pipeline2 == None:
+        ...
+
+    pipeline3 = make_pipeline(StandardScaler(), LinearDiscriminantAnalysis()) # Noncompliant
+    def a_call(): ...
+    def b_call(): ...
+    a_call(b_call(pipeline3))
```
[the remark this answers is not in the document]
Comment (ghislainpiot) [reply python-checks/src/test/resources/checks/sklearn_pipeline_specify_memory_argument.py:142]: We only raise an issue on the inner `pipe`, which is the expected behavior

**Review by joke1196 — APPROVED** [review]
LGTM! Thanks for taking of the refactoring

**Pull request merged**